[WebGPU] Remove size member from GPUCanvasConfiguration

tuankiet65 opened the pull request:
#### 68e829f4c00235e0f8e890783f5ec8ab6a14c7dc
<pre>
[WebGPU] Remove size member from GPUCanvasConfiguration
<a href="https://bugs.webkit.org/show_bug.cgi?id=242003">https://bugs.webkit.org/show_bug.cgi?id=242003</a>

Reviewed by Myles C. Maxfield.

GPUCanvasConfiguration::size was removed in
<a href="https://github.com/gpuweb/gpuweb/pull/2826">https://github.com/gpuweb/gpuweb/pull/2826</a> in favor of
using the canvas size directly.

* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.h:
(WebCore::GPUCanvasConfiguration::convertToBacking const):
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUCanvasConfiguration.h:
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.cpp:
(WebKit::WebGPU::ConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::ConvertFromBackingContext::convertFromBacking):
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.h:
(WebKit::WebGPU::CanvasConfiguration::encode const):
(WebKit::WebGPU::CanvasConfiguration::decode):

Canonical link: <a href="https://commits.webkit.org/251882@main">https://commits.webkit.org/251882@main</a>
</pre>

litherum commented on Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.h:
```diff
@@ -51,8 +47,7 @@ struct GPUCanvasConfiguration {
             WebCore::convertToBacking(colorSpace),
-            WebCore::convertToBacking(compositingAlphaMode),
-            size ? std::optional { WebCore::convertToBacking(*size) } : std::nullopt,
+            WebCore::convertToBacking(compositingAlphaMode)
```
Can we keep the trailing comma?

tuankiet65 replied:
Done.

tuankiet65 committed:
[WebGPU] Remove size member from GPUCanvasConfiguration
https://bugs.webkit.org/show_bug.cgi?id=242003

Reviewed by Myles C. Maxfield.

GPUCanvasConfiguration::size was removed in
https://github.com/gpuweb/gpuweb/pull/2826 in favor of
using the canvas size directly.

* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.h:
(WebCore::GPUCanvasConfiguration::convertToBacking const):
* Source/WebCore/Modules/WebGPU/GPUCanvasConfiguration.idl:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUCanvasConfiguration.h:
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.cpp:
(WebKit::WebGPU::ConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::ConvertFromBackingContext::convertFromBacking):
* Source/WebKit/Shared/WebGPU/WebGPUCanvasConfiguration.h:
(WebKit::WebGPU::CanvasConfiguration::encode const):
(WebKit::WebGPU::CanvasConfiguration::decode):

Canonical link: https://commits.webkit.org/251882@main
